### ISSUE TITLE
Select line/yank/paste inconsistency

### ISSUE THREAD

**Issue by chapmandu** [jasonwilliams/vscode-helix#18]
Give this file content:
```
<div data-foo="bar">
  baz
</div>
```

Placing the cursor on the 2nd (baz) line and using `xyp` (select line - yank - paste) should duplicate the "baz" line.

Helix:
![image](https://github.com/jasonwilliams/vscode-helix/assets/1092563/e1ed7396-55e2-43c9-9c07-f52df7352a24)

VSCode:
![image](https://github.com/jasonwilliams/vscode-helix/assets/1092563/eafb8f9e-3fa3-4823-9e10-7e656960c324)

**Comment by jasonwilliams** [comment]
Yeah its bad and needs looking at, the implementation behind that needs starting again really. I should look how vscode vim are handling it. If you have any spare capacity feel free to look at this, i may not get around to it for a while.

**Comment by jasonwilliams** [comment]
One option which I think should solve this for line copies.

When you do x and then y, I may just defer to VS Code's line copy and when you do paste just use their line paste.

It should fix the problem above but won't fix block copy, that still needs some help

**Comment by jasonwilliams** [comment]
Seeing that VSCode/Vim have fixed most of these issues I've decided to fork the modern version and add helix features back in. You can follow the branch here: https://github.com/jasonwilliams/vscode-helix/compare/helix-vim-fork?expand=1

It's a bit of a start-again approach but it will be better than re-inventing the wheel when most of these edge cases have already been dealt with

**Comment by chapmandu** [comment]
Hey Jason, what would you like help with (I've little experience with TypeScript).. I built and installed this branch, though I'm not sure in what state the keybinding are?

**Comment by jasonwilliams** [comment]
Hey @chapmandu thanks for helping!

So I've decided to fork vscodeVim because that has a lot of functionality we need. We can then tweak the bits it doesn't have, but we should be able to get 90% of the way there with their code base.

Feel free to make PRs to the branch I linked above, or if you have a more fresh fork (I think I need to revert some changes in that branch) feel free to push it as a new branch.

**Comment by chapmandu** [comment]
Righto..I had a quick browse through the docs I think I can make a start.. But how did you plan on implementing Helix keybindings? Did you want to add a `keybinding.json`, modify the `package.json` or modify the `/src/actions/commands/actions.ts`?

Any pointers appreciated..

**Comment by jasonwilliams** [comment]
Ok so this is what you need to do.
1. checkout the branch helix-fork (https://github.com/jasonwilliams/vscode-helix/tree/helix-fork)
2. build this branch, `npm i, npm run build`
3. To change a keybinding just search that keybinding, change it then rebuild again
4. use the debugger to test changes

So for example https://github.com/VSCodeVim/Vim/blob/46436e90f3101d3dde7779183a0fbd02750089d0/src/actions/commands/actions.ts#L914-L936 here you would change 'V' to 'x'
